L1Trigger/Phase2L1ParticleFlow/src/PFAlgoBase.cc

```diff
@@ -150,31 +150,64 @@ void PFAlgoBase::doVertexing(std::vector<Region> &rs, VertexAlgo algo, float &pv
-                                float sum3 = h_dz->GetBinContent(b) + h_dz->GetBinContent(b+1) + h_dz->GetBinContent(b-1);
-                                if (bmax == -1 || sum3 > max) { max = sum3; bmax = b; }
-                            }
-                            pvdz = h_dz->GetXaxis()->GetBinCenter(bmax); 
-                        }; break;
-    }
+    // switch(algo) {
+    //     case OldVtxAlgo: {
+    //                          int imaxbin = h_dz->GetMaximumBin();
+    //                          pvdz = h_dz->GetXaxis()->GetBinCenter(imaxbin);
+    //                      }; break;
+    //     case TPVtxAlgo: {
+    //                         float max = 0; int bmax = -1;
+    //                         for (int b = 1; b <= lNBins; ++b) {
+    //                             float sum3 = h_dz->GetBinContent(b) + h_dz->GetBinContent(b+1) + h_dz->GetBinContent(b-1);
+    //                             if (bmax == -1 || sum3 > max) { max = sum3; bmax = b; }
+    //                         }
+    //                         pvdz = h_dz->GetXaxis()->GetBinCenter(bmax); 
+    //                     }; break;
+    //     case StandaloneVtxProd: {
+    //       // extract the vertex information from the edm file
+    //       // 
+    //                             }; break;
+
+    // }
+
+    // std::cout << "Calculated z0 is " << pvdz
+    //           << std::endl;
+
+    // extract vertex from edm file
+
+    // -------------------------------------------------------------------------
+
+    // std::cout << "meh attaching tracks to tdr vertex - PUPPI" << std::endl;
+
+    // int vtx_tracks = 0;
+    // int n_tracks = 0;
+
     int16_t iZ0 = round(pvdz * InputTrack::Z0_SCALE);
-    int16_t iDZ  = round(1.5 * vtxRes_ * InputTrack::Z0_SCALE);
+    int16_t iDZ  = round(1 * vtxRes_ * InputTrack::Z0_SCALE);
     int16_t iDZ2 = vtxAdaptiveCut_ ? round(4.0 * vtxRes_ * InputTrack::Z0_SCALE) : iDZ;
+
+    // std::cout << "meh: iDZ = " << iDZ << ", iDZ2 = " << iDZ2 << ", scale = "
+    //           << InputTrack::Z0_SCALE << std::endl;
+
     for (Region & r : rs) {
         for (PropagatedTrack & p : r.track) {
             bool central = std::abs(p.hwVtxEta) < InputTrack::VTX_ETA_1p3;
             if (r.relativeCoordinates) central = (std::abs(r.globalAbsEta(p.floatVtxEta())) < 1.3); // FIXME could make a better integer implementation
             p.fromPV = (std::abs(p.hwZ0 - iZ0) < (central ? iDZ : iDZ2));
+            // if (std::abs(p.hwZ0 - iZ0) < (central ? iDZ : iDZ2)) {
+            //   std::cout << "meh: attaching track to tdr vertex - "
+            //             << "pT = " << p.hwPt
+            //             << ", hwEta = " << p.hwEta << ", hwPhi = " << p.hwPhi
+            //             << std::endl;
+            //   vtx_tracks++;
+            // }
+            // n_tracks++;
         }
     }
 
+    // std::cout << "no tracks attached = " << vtx_tracks
+    //           << "/" << n_tracks
+    //           << std::endl;
```

Comment:
also these two commented pieces could probably be cleaned up in the final version (or collapse them in a single line)